cmd/spire-server/cli/run/run.go

```diff
@@ -406,33 +427,36 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 		}
 
 		federatesWith := map[spiffeid.TrustDomain]bundleClient.TrustDomainConfig{}
-		for trustDomain, config := range c.Server.Federation.FederatesWith {
-			port := defaultBundleEndpointPort
-			if config.BundleEndpoint.Port != 0 {
-				port = config.BundleEndpoint.Port
-			}
-			if config.BundleEndpoint.UseWebPKI && config.BundleEndpoint.SpiffeID != "" {
-				return nil, errors.New("usage of `bundle_endpoint.spiffe_id` is not allowed when authenticating with Web PKI")
-			}
-
-			var spiffeID spiffeid.ID
-			if config.BundleEndpoint.SpiffeID != "" {
-				spiffeID, err = spiffeid.FromString(config.BundleEndpoint.SpiffeID)
-				if err != nil {
-					return nil, err
-				}
-			}
 
+		for trustDomain, config := range c.Server.Federation.FederatesWith {
 			td, err := spiffeid.TrustDomainFromString(trustDomain)
 			if err != nil {
 				return nil, err
 			}
 
-			federatesWith[td] = bundleClient.TrustDomainConfig{
-				EndpointAddress:  fmt.Sprintf("%s:%d", config.BundleEndpoint.Address, port),
-				EndpointSpiffeID: spiffeID,
-				UseWebPKI:        config.BundleEndpoint.UseWebPKI,
+			var trustDomainConfig *bundleClient.TrustDomainConfig
+			switch {
+			case config.DeprecatedBundleEndpoint != nil:
+				sc.Log.Warn("The `bundle_endpoint` configurable is deprecated and will be removed in a future release. Please use `bundle_endpoint_url` and `bundle_endpoint_profile` to configure the federation relationships instead.")
```

Comment:
I think we need to be a little more specific here since `bundle_endpoint` is also used at federation top level for our own bundle endpoint config

```suggestion
				sc.Log.Warn("The `bundle_endpoint` configurable inside `federates_with` is deprecated and will be removed in a future release. Please use `bundle_endpoint_url` and `bundle_endpoint_profile` to configure the federation with %q instead.", trustDomain)
```